pkg/git/errors.go

```diff
@@ -0,0 +1,40 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+type gitError struct {
+	msg string
+}
+
+// GitError returns an error whereby the user part of the url is removed (e.g. an access token)
+// Accepts the message to use and the url to remove the user part from
+func GitError(msg, url string) error {
+	cleanedURL, err := cleanURL(url)
+	if err != nil {
+		return err
+	}
+	return gitError{msg: fmt.Sprintf("%s. repository URL = %s", msg, cleanedURL)}
+}
+
+// IsGitError checks if the error is a Git one (so there'll be a .User field in the URL)
```

Comment:
This comment could be expanded to explain how the presence or lack of a .User field relates to what the code is doing.

Reply:
Still trying to get this to ever be true at the moment...

Reply:
@bigkevmcd hey Kevin, could you share some insights please? So AIUI we're doing a cast to our gitError type struct, what I don't get is how isGitError can ever be true, have I missed a step implementing this? Still tryna figure it out actually, thought I had it!

@mnuttall small unit tests added for CleanURL

Reply:
Regarding

> 
What's the latest on this @a-roberts ? Is err unsafe to print at this point?


for 

```
func addCredentialsIfNecessary(repoURL string, author *git.Author) (string, error) {
	parsed, err := url.Parse(repoUrl)
	if err != nil {
```

I've determined it is unsafe to print, so if I change that Url.parse to accept `:heyIamyourtoken@github.com` and *not* repoURL, the error that would come out (if we just print the error) is

```
parse ":heymarkIamyourtoken@github.com": missing protocol scheme
```

so I think I'm right to go with the omission of it. It's worth noting I could only get to this path by hardcoding `:heymarkIamyourtoken@github.com` for the parse method to use, to get the error out. I'll add a comment in the code to make that clear.